privaterelay/apps.py

```diff
@@ -29,30 +30,44 @@ def get_profiler_startup_data() -> tuple[str | None, str | None]:
     return service, version
 
 
+def write_gcp_key_json_file(gcp_key_json_path: Path) -> None:
+    # create the gcp key json file from contents of GOOGLE_CLOUD_PROFILER_CREDENTIALS_B64
+    google_app_creds = base64.b64decode(settings.GOOGLE_CLOUD_PROFILER_CREDENTIALS_B64)
+    if not google_app_creds == b"":
+        with open(gcp_key_json_path, "w+") as gcp_key_file:
+            gcp_key_file.write(google_app_creds.decode("utf-8"))
+
+
 class PrivateRelayConfig(AppConfig):
     name = "privaterelay"
 
     def ready(self) -> None:
-        if settings.GOOGLE_APPLICATION_CREDENTIALS:
+        if (
+            settings.GOOGLE_APPLICATION_CREDENTIALS != ""
+            and settings.GOOGLE_CLOUD_PROFILER_CREDENTIALS_B64 != ""
+        ):
             # Set up Google Cloud Profiler
             service, version = get_profiler_startup_data()
             if service != None:
-                import googlecloudprofiler
-
-                # Make sure the expect gcp_key.json file exists
                 gcp_key_json_path = Path(settings.GOOGLE_APPLICATION_CREDENTIALS)
-                if gcp_key_json_path.exists() and gcp_key_json_path.is_file():
+                if not gcp_key_json_path.exists() or not gcp_key_json_path.is_file():
```

Comment:
If you set `GOOGLE_APPLICATION_CREDENTIALS` to `"privaterelay"`, this fails with `IsADirectoryError: [Errno 21] Is a directory: 'privaterelay'`. I think the line should be:

```suggestion
                if not gcp_key_json_path.exists() and gcp_key_json_path.is_file():
```

At the same time, you'll get a `IsADirectoryError` in a couple of lines when you try to read it. So, maybe drop the `is_file()` check and let Relay crash on startup.

Reply:
I removed the `is_file` check and expanded the `try/except` block around the code for the file so that the Relay app can still start up and run even if google cloud profiler can't.